src/Fixer/Whitespace/StatementIndentationFixer.php

```diff
@@ -0,0 +1,604 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Analyzer\AlternativeSyntaxAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class StatementIndentationFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    private $bracesFixerCompatibility;
+
+    public function __construct(bool $bracesFixerCompatibility = false)
+    {
+        parent::__construct();
+
+        $this->bracesFixerCompatibility = $bracesFixerCompatibility;
```

Comment:
i hardcoded `false` as value and run tests for `BracesFixer`. I actually like the changes applied with `StatementIndentationFixer`. Maybe we should drop this "compatibility" mode at all and apply changes introduces by this Fixer to `BracesFixer` as bugfixes?

Reply:
The new behavior when using `false` is described in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4884#discussion_r399457536. I'm completely ok to drop the "braces compatibility" mode entirely but that might be considered a BC break as `braces` behavior would change. Is everyone ok with that?

Reply:
I don't know where to discuss it, so I will put it in this PR

> 3. In statement_indentation: when a comment looks like commented-out code, don't indent it

OK, I agree it could make value for folks who are committing commented code or who are commenting out code and run the fixer run while that. Maybe we should keep behaviour of BracesFixer (for both fixers, old and new, without config)

> 4. In statement_indentation: when a single line comment is part of a multiline message, don't indent it

sounds legit, right? let's keep behaviour of BracesFixer (for both fixers, old and new, without config)

> 5. In statement_indentation: assume code that is not inside any curly braces block is correctly indented

that was actually all I saw when I applied rule with hardcoded `false` (probably due to amount of cases when it was applied, compared to points 3 and 4).

it was not assuming that. it was mine "simplification" when I was creating `BracesFixer`, as somehow I was struggling to apply the logic to code on root level (not nested in curly braces block). If both fixers would now be able to handle root-level code, it's great (for both fixers, old and new, without config)

Reply:
> OK, I agree it could make value for folks who are committing commented code or who are commenting out code and run the fixer run while that. Maybe we should keep behaviour of BracesFixer (for both fixers, old and new, without config)

Since version control systems are widespread, I consider commenting out code useless and a bad practice. We should not support that anymore IMO. Having those comments indented can even help spotting commented-out code that should be removed.

> sounds legit, right? let's keep behaviour of BracesFixer (for both fixers, old and new, without config)

Meh. I never saw such case, I assume it's very rare.

---
So, we both want to drop 5, but you want to keep 3 and 4 while I don't. Any other opinion? :)

Reply:
> 3

agree about commiting it, but I can imagine running the Fixer during development, eg automatically by CI on change, and having the code pushed few tabs to right can be distracting

> 4

I saw it few times, IIRC we added it because someone was raising this topic

------------------

(happy to hear more opinions as well)

Reply:
> 5. shouldn't all code outside braces block be un-indented by this fixer?

@SpacePossum Yes, that's what the new `statement_indentation` rule does, unless it's ran via `braces` to keep its current behavior. Should I understand that you are in favor of applying this new behavior to `braces`? If so, looking at the discussion, the consensus seems to be:
* indent comments even if they look like commented-out code (dropping specific behavior of `braces`);
* keep indenting single line comments that form multiline messages (importing specific behavior from `braces` into `statement_indentation`);
* always indent code outside braces (dropping specific behavior of `braces`).

Which means I could remove the `$bracesFixerCompatibility` parameter entirely. Is everyone ok with this?